### PR TITLE
Added preprocessing logic; because the 'all_objects' function current…

### PR DESCRIPTION
**Pull request by DebjyotiRay**
[ENH] Add filter_tags preprocessing to accept strings and lists in all_objects and get_package_metadata

#### Reference Issues
 Fixes #8210 from sktime

#### Changes:
This PR enhances the all_objects and get_package_metadata functions by adding preprocessing logic for the filter_tags parameter. Previously, these functions only handled filter_tags as a dict. Now they accept more user-friendly input formats:
 i) String input: `filter_tags="A"` → converts to `{"A": True}`
 ii) List/tuple input: `filter_tags=["A", "B"]` → converts to `{"A": True, "B": True}`
 iii) Dict input: Existing functionality preserved (dict is copied to prevent modification)
 iv) Error handling: Clear TypeError for invalid input types

#### What should a reviewer concentrate their feedback on?
- [x] Preprocessing logic: Verify the string/list-to-dict conversion works correctly
- [x] Backward compatibility: Ensure existing dict-based usage continues to work unchanged.

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

Also, please do not change the logic of the existing tests.
Changing the error message to match the argument is of course ok.

**Comment by DebjyotiRay** [comment]
Hello!
I tried to keep the logic of most of the existing tests intact.
Only changed some input and output formats of the tests corresponding to my changes, and added fresh ones for my edge cases.

Please let me know your review on them.

File: skbase/lookup/_lookup.py [hunk 1]
```diff
@@ -190,34 +190,23 @@ def _filter_by_tags(obj, tag_filter=None, as_dataframe=True):
     if tag_filter is None:
         return True
 
-    type_msg = (
```
Comment (fkiraly) [inline skbase/lookup/_lookup.py:193]: it looks like the function already allowed for iterable of str or str

**Comment by DebjyotiRay** [comment]
Hello @fkiraly !
I updated the changes. Please lmk your thoughts.

File: skbase/lookup/_lookup.py (hunk 1 above)
Comment (fkiraly) [reply skbase/lookup/_lookup.py:193]: it seems like you are still changing the logic itself, making it less general, and removing the useful error message.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

It seems like you are still changing the logic itself, making it less general, and removing the useful error message.

Is this really necessary? Please comment.
